### PR TITLE
Replace webpack with rollup

### PR DESCRIPTION
**Pull request by SidKwok**
## It is an update for tools.

Webpack is great in managing a whole project, but rollup seems better for a library.

Bundle sizes are reduced:

```
axios.js:     42.6 kb -> 37.5 kb
axios.min.js: 12.9 kb -> 10.8 kb
```

I ran the tests and it seems fine, but some weird faults came up no matter which bundler I use:
```
Using tsc v2.5.2
07 09 2017 12:53:43.712:WARN [Safari 10.1.2 (Mac OS X 10.12.6)]: Disconnected (1 times)
Safari 10.1.2 (Mac OS X 10.12.6) ERROR
  Disconnectedundefined
Safari 10.1.2 (Mac OS X 10.12.6): Executed 8 of 179 DISCONNECTED (13.245 secs / 0.68 secs)
07 09 2017 12:53:43.713:ERROR [reporter.sauce]: ✖ Test Disconnected
07 09 2017 12:53:43.714:INFO [karma]: Restarting Safari 10.1.2 (Mac OS X 10.12.6) (1 of 1 attempts)
07 09 2017 12:53:43.718:WARN [Safari 10.1.2 (Mac OS X 10.12.6)]: Disconnected (1 times)
Safari 10.1.2 (Mac OS X 10.12.6) ERROR
  Disconnectedundefined
Safari 10.1.2 (Mac OS X 10.12.6): Executed 0 of 0 DISCONNECTED (15.954 secs / 0 secs)
07 09 2017 12:53:43.718:ERROR [reporter.sauce]: ✖ Test Disconnected
07 09 2017 12:53:43.718:INFO [karma]: Restarting Safari 10.1.2 (Mac OS X 10.12.6) (1 of 1 attempts)
```

But it should make no side effects to the bundles, I guess.

And I also ran `example` and it works just fine.

Let me know if you need more information or support.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,11 +61,17 @@
...
+    "rollup-plugin-node-resolve": "^3.0.0",
+    "rollup-plugin-replace": "^2.0.0",
+    "rollup-plugin-uglify": "^2.0.1",
     "sinon": "^1.17.4",
-    "webpack": "^1.13.1",
-    "webpack-dev-server": "^1.14.1",
+    "typescript": "^2.0.3",
     "url-search-params": "^0.6.1",
-    "typescript": "^2.0.3"
+    "webpack": "^1.13.1",
```
Comment (Jamesking56) [inline package.json]: If we're replacing with rollup, why do we still need webpack as a dependency?

**Comment by SidKwok** [comment]
@JustinBeckwith I will be happy to help, and may commit a PR in couple days.

I see the major conflicting files are `./dist` and `package.json`, should I overwrite the dist files in next commit?

**Comment by JustinBeckwith** [comment]
Sweet 🍬 I'm still pretty new around these parts, but I think you're supposed to leave dist/ changes out of this according to the contributing guide:
https://github.com/axios/axios/blob/master/CONTRIBUTING.md#developing

Probably best just to drop those changes!

**Comment by SidKwok** [comment]
ok, I will commit these changes after I read the guide.

And I'm confusing about one thing in [`krama.conf.js`](https://github.com/axios/axios/blob/master/karma.conf.js#L195), why `var undefined`?

**Comment by AllainPL** [comment]
I am worried that after 1,5 years this might be just obsolete ;) We have now rollup 1.9.3 instead of 0.49.2 so i would probably reject it by now @emilyemorehouse. Also it might be up for discussion what is now better to use webpack 4 or rollup.

**Comment by yasuf** [comment]
This is sort of stale now, sorry this has to be rejected but the reduction in size of the build looked pretty attractive. Thanks everyone, I'm starting to take a look at PRs and either merging or closing them

**Pull request closed without merging**